libs/dyn/convert/from_typed.go

```diff
@@ -99,8 +108,13 @@ func fromTypedStruct(src reflect.Value, ref dyn.Value) (dyn.Value, error) {
 			refv = dyn.NilValue
 		}
 
+		var options []fromTypedOptions
+		if v.Kind() == reflect.Interface {
+			options = append(options, includeZeroValuedScalars)
+		}
+
```

Comment:
This is to capture a variable value equal to `0`, right?

Previously this would have been done through the non-nil pointer and dereferencing that.